File: src/main/java/niklasu/speedtester/downloader/DownloadScheduler.java
package niklasu.speedtester.downloader;

import static java.util.concurrent.TimeUnit.MINUTES;
import static java.util.concurrent.TimeUnit.SECONDS;

import niklasu.speedtester.config.ConfigProvider;

import java.util.concurrent.Executors;

public class DownloadScheduler {

    private final DownloadThread downloadThread;
    private final ConfigProvider configProvider;

    public DownloadScheduler(DownloadThread downloadThread, ConfigProvider configProvider) {

        this.downloadThread = downloadThread;
        this.configProvider = configProvider;
    }

    public void start() {
        if (configProvider.interval() == 0) {
            downloadThread.start();
        } else {
            Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(
                    downloadThread,
                    0,
                    configProvider.interval(),
                    SECONDS);
        }
    }
}

File: src/main/java/niklasu/speedtester/downloader/DownloadScheduler.kt
package niklasu.speedtester.downloader

import com.google.inject.Inject
import niklasu.speedtester.config.ConfigProvider
import java.util.concurrent.Executors
import java.util.concurrent.TimeUnit.MINUTES


class DownloadScheduler @Inject
constructor(private val downloadThread: DownloadThread, private val configProvider: ConfigProvider) {
    fun start() {
        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(
                downloadThread,
                0,
                configProvider.interval.toLong(),
                MINUTES)
    }
}
